src/Orleans/Utils/Factory.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Orleans
+{
+    /// <summary>
+    /// Creates an instance of <typeparamref name="TInstance"/>.
+    /// </summary>
+    /// <typeparam name="TInstance"></typeparam>
+    /// <returns></returns>
+    public delegate TInstance Factory<out TInstance>();
```

Comment:
yeah, naming-wise I don't mind, at some point I was going to also suggest calling it `ObjectFactory<T>`, but since it's a different shape altogether, I prefer not to be confused with it. Still, I wouldn't mind a different name either if we want to have something less ambiguous than `Factory<T>`, but I'd vote no for `ObjectFactory<T>` because of that reason